job/boltjobdb_test.go

```diff
@@ -22,27 +26,35 @@ func TestSaveAndGetJob(t *testing.T) {
 	assert.Equal(t, j.Schedule, genericMockJob.Schedule)
 	assert.Equal(t, j.Owner, genericMockJob.Owner)
 	assert.Equal(t, j.SuccessCount, genericMockJob.SuccessCount)
+
+	cache.jobDB = nil
 }
 
 func TestDeleteJob(t *testing.T) {
+	db := GetBoltDB(testDbPath)
+	cache := NewMemoryJobCache(db, time.Second*60)
+	defer db.Close()
+
 	genericMockJob := getMockJobWithGenericSchedule()
-	genericMockJob.Init()
-	genericMockJob.Save()
-	AllJobs.Set(genericMockJob)
+	genericMockJob.Init(cache)
+	genericMockJob.Save(db)
+	cache.Set(genericMockJob)
 
 	// Make sure its there
-	j, err := GetJob(genericMockJob.Id)
+	j, err := db.Get(genericMockJob.Id)
 	assert.Nil(t, err)
 	assert.Equal(t, j.Name, genericMockJob.Name)
-	assert.NotNil(t, AllJobs.Get(genericMockJob.Id))
+	assert.NotNil(t, cache.Get(genericMockJob.Id))
 
 	// Delete it
-	genericMockJob.Delete()
+	genericMockJob.Delete(cache, db)
 
-	k, err := GetJob(genericMockJob.Id)
+	k, err := db.Get(genericMockJob.Id)
 	assert.Error(t, err)
 	assert.Nil(t, k)
-	assert.Nil(t, AllJobs.Get(genericMockJob.Id))
+	assert.Nil(t, cache.Get(genericMockJob.Id))
+
+	genericMockJob.Delete(cache, db)
 
-	genericMockJob.Delete()
+	cache.jobDB = nil
```

Comment:
@ajvb Are you doing this to clear out the test's state at the end? Check out some of testify/suite's teardown methods to just have this happen automatically.

If there's another reason, then at least add a comment :)